### PR TITLE
Add ADR for build tool

### PR DESCRIPTION
**Pull request by emmyoop**
resolves https://github.com/dbt-labs/dbt-adapter/issues/3

### Description

This is just the adr to make it more clear.  #21 will follow with more complete hatch implementation.

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-common/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-common/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Review by martynydbt — COMMENTED** [review]
Having only one maintainer is pretty high risk. 
How much effort does it take to switch between tools should it become abandonware?

**Comment by emmyoop** [comment]
@martynydbt while it only has 1 maintainer it is actually part of the Python Packaging Authority (PyPA) and lives in the PyPA org in GitHub.  PyPA is a working group that maintains a core set of software projects used in Python packaging.  it also has had 69 contributors over its lifetime.  I can add this color to the doc.

File: docs/arch/adr-0001-build-tooling.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+- `poetry`
+
+
+### setuptools
+
+#### Pro's
+
+- most popular option
+- supported by Python Packaging Authority
+- build tool of record for existing internal adapters
```
Comment (emmyoop) [inline docs/arch/adr-0001-build-tooling.md:22]:
```suggestion
- build tool of record for dbt-core & existing internal adapters
```

File: docs/arch/adr-0001-build-tooling.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- supports configuring workflows in `pyproject.toml` (same functionality as `make`)
+- incorporates new PEP's quickly
+- Manages python distributions itself without need of pyenv.  This allows Windows and non-Windows users to both work locally in the same way.
+- used by black, tox, pipx, Jupyter Notebook, Datadog
+
+#### Con's
+
+- far less popular than other options
+- no dependency management (manually add to `pyproject.toml`)
+- only one maintainer (but is part of the larger PyPA working group)
```
Comment (emmyoop) [inline docs/arch/adr-0001-build-tooling.md:46]:
```suggestion
- only one maintainer (but is officially part of the larger PyPA working group)
```

File: docs/arch/adr-0001-build-tooling.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+This option aligns with `dbt-adapter` and `dbt-semantic-layer`, which minimizes confusion
+for anyone working in multiple repositories.
+`hatch` also replaces `tox` and `make`, which consolidates our toolset to make working locally and with CI more consistant.
+
+
+## Consequences
+
+- [+] retire `tox`
+- [+] retire `make`
```
Comment (emmyoop) [inline docs/arch/adr-0001-build-tooling.md:76]:
```suggestion
- [+] retire `make`
- [+] rewriting the release workflows will create a more intuitive release for hatch projects
- [-] we cannot reuse the existing release workflows
```

File: docs/arch/adr-0001-build-tooling.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+- incorporates new PEP's quickly
+- Manages python distributions itself without need of pyenv.  This allows Windows and non-Windows users to both work locally in the same way.
+- used by black, tox, pipx, Jupyter Notebook, Datadog
+
+#### Con's
+
+- far less popular than other options
+- no dependency management (manually add to `pyproject.toml`)
+- only one maintainer (but is officially part of the larger PyPA working group)
+- Hatch does not allow for the installation of specific patch release versions but rather only uses minor release granularity that tracks the latest patch release
```
Comment (emmyoop) [inline docs/arch/adr-0001-build-tooling.md:47]:
```suggestion
- Hatch does not allow for the installation of specific patch release versions of itself but rather only uses minor release granularity that tracks the latest patch release
```

**Review by MichelleArk — APPROVED** [review]

**Pull request merged**